### ISSUE TITLE
Optional parts not working as expected

### ISSUE THREAD

**Issue by fanste** [RobinHerbots/jquery.inputmask#2710]
I have an mask containing optionals, like `*-******[-**]` and `clearIncomplete = true`.

My expectation is, that it either accepts `a-bcdefg` or `a-bcdefg-hi` but NOT `a-bcdefg-h` (last character missing).

Is this a bug or a desired behavior?
Tested using the latest version from `dist`.

**Edited on 2023-05-22**.

**Comment by RobinHerbots** [comment]
Can you retry with 5.0.9-beta.7

**Comment by fanste** [comment]
Same behavior as before.

**Comment by RobinHerbots** [comment]
@fanste ,

I tried your mask.
Initial: _-______
Type: abcdefgh => a-bcdefg-h_
Blur: a-bcdefg-h_

The mask is visually ok, but I noticed a oncomplete is triggered, which is incorrect.
Is this what you mean?

**Comment by fanste** [comment]
I noticed, that I missed a detail in the initial posting, sorry for that.
I've edited the initial posting and added the info about **`clearIncomplete = true`**.


So in other words:

- The InputMask is configured to clear the entered value, if the mask is not fully filled ( => `clearIncomplete=true`)
- The mask is set to `*-******[-**]`
- If I only fill `1-234567-8` (missing the last position) and blur, what is the desired outcome?
   1. clear the field, as the input is incomplete and `clearIncomplete` is set
   2. keep the value as it is currently implemented

You mentioned that `oncomplete` is triggered which should not be the case. So i think that _clear the field_ (=> option 1) should be the desired behavior?
